apps/src/codegen.js

```diff
@@ -3,6 +3,15 @@
 var dropletUtils = require('./dropletUtils');
 var utils = require('./utils');
 
+/** @const {number} */
+exports.FOR_STATEMENT_MODE_INIT = 0;
+/** @const {number} */
+exports.FOR_STATEMENT_MODE_TEST = 1;
+/** @const {number} */
+exports.FOR_STATEMENT_MODE_BODY = 2;
+/** @const {number} */
+exports.FOR_STATEMENT_MODE_UPDATE = 3;
```

Comment:
I very much approve of having nice English descriptions for these instead of magic numbers :) However, I would prefer to have these as an enum, i.e.:
```
exports.ForStatementMode = {
  INIT: 0,
  TEST: 1,
  BODY: 2,
  UPDATE: 3
};